src/OrbitDB.js

```diff
@@ -59,19 +75,25 @@ let databaseTypes = {
 
     const { id } = await ipfs.id()
     const directory = options.directory || './orbitdb'
-    const keystore = options.keystore || Keystore.create(path.join(directory, id, '/keystore'))
+    const storageOptions = Object.assign({}, options.storage, {
+      keystore: await storage.createStore(path.join(directory, id, '/keystore')),
```

Comment:
Could this be made a function which is only called if a keystore isn't passed? Currently this store will be created regardless of whether a keystore is passed. Also if the passed keystore uses the same path we will get the `LOCK: held by other process` error. What do you think?

Reply:
Yes this is probably my least favorite part of this refactor. I think your idea is sound. Let me see what I can come up with!

Reply:
Have a look! I think it looks pretty good now.

Reply:
How's that look?

Reply:
Lgtm! :+1: